### PR TITLE
Fixed `set_period()` not updating decomposer parameters

### LINKED ISSUE #3771: Decomposer does not update parameters dictionary when set_seasonal_period is called.
There's some weird gotcha with the parameters dictionary in the Decomposer class that needs to be fixed so that `set_seasonal_period` can set that parameter dictionary correctly.

#### Code Sample, a copy-pastable example to reproduce your bug.

```python
from evalml.pipelines.components.transformers.preprocessing.stl_decomposer import STLDecomposer

stl = STLDecomposer()
stl.fit(X,y)

stl.set_seasonal_period(X,y)

assert stl.parameters["seasonal_period"] != -1 # the default
```

### PR DESCRIPTION
**Pull request by christopherbunn**
Also introduces a way to update parameters in a component.

Resolves #3771

### CODE REVIEW

File: evalml/pipelines/components/transformers/preprocessing/decomposer.py [hunk 1]
```diff
@@ -205,7 +205,7 @@ def set_period(self, X: pd.DataFrame, y: pd.Series):
 
         """
         self.period = self.determine_periodicity(X, y)
-        self.parameters.update({"period": self.period})
+        self.update_parameters({"period": self.period})
```
Comment (jeremyliweishih) [inline evalml/pipelines/components/transformers/preprocessing/decomposer.py:208]: can you add a test verifying the period updates?

**Review by jeremyliweishih — APPROVED** [review]
LGTM but would be nice to have an explicit test for the decomposer.

File: evalml/pipelines/components/transformers/preprocessing/decomposer.py (hunk 1 above)
Comment (eccabay) [reply evalml/pipelines/components/transformers/preprocessing/decomposer.py:208]: There's a line in `test_decomposer.py::test_decomposer_set_period` that's currently commented out testing this, you should be able to uncomment that!

**Review by eccabay — APPROVED** [review]
Looks good! I agree with Jeremy about the testing. Have you verified if there's anywhere else we update parameters that could benefit from this function?

File: evalml/pipelines/components/transformers/preprocessing/decomposer.py (hunk 1 above)
Comment (christopherbunn) [reply evalml/pipelines/components/transformers/preprocessing/decomposer.py:208]: I uncommented that line, I think that should be sufficient? @jeremyliweishih I can also break it out into its own separate test if need be.

**Comment by christopherbunn** [comment]
@eccabay I did a quick ctrl+f search for places that update parameters and this should be it. I think this is the only place where were updating the parameters from inside the component itself.

File: evalml/pipelines/components/transformers/preprocessing/decomposer.py (hunk 1 above)
Comment (jeremyliweishih) [reply evalml/pipelines/components/transformers/preprocessing/decomposer.py:208]: @christopherbunn thats great!

**Pull request merged**